src/libcore/hash/sip.rs

```diff
@@ -30,22 +48,31 @@ use super::Hasher;
 /// Although the SipHash algorithm is considered to be generally strong,
 /// it is not intended for cryptographic purposes. As such, all
 /// cryptographic uses of this implementation are _strongly discouraged_.
-#[derive(Debug)]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub struct SipHasher {
+#[derive(Debug, Clone, Default)]
+pub struct SipHasher(SipHasher24);
```

Comment:
It's not a thing we've ever really worried about since code that actually contains that seems a bit unlikely, but the compiler should really accept any of `let Foo(...) = expression` and `let Foo { ... } = expression` work for a fully opaque type.